hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImplV2.java

```diff
@@ -0,0 +1,412 @@
+          LOG.warn("Could not commit delete block transaction: " +
+              transactionResult.getTxID(), e);
+        }
+      }
+      try {
+        deletedBlockLogStateManager
+            .removeTransactionsFromDB(transactionIDs.build());
+      } catch (IOException e) {
+        LOG.warn("Could not commit delete block transactions: " +
+            transactionIDs.build().getTxIDList(), e);
+      }
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  private boolean isTransactionFailed(DeleteBlockTransactionResult result) {
+    if (LOG.isDebugEnabled()) {
+      LOG.debug(
+          "Got block deletion ACK from datanode, TXIDs={}, " + "success={}",
+          result.getTxID(), result.getSuccess());
+    }
+    if (!result.getSuccess()) {
+      LOG.warn("Got failed ACK for TXID={}, prepare to resend the "
+          + "TX in next interval", result.getTxID());
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @param containerID - container ID.
+   * @param blocks      - blocks that belong to the same container.
+   * @throws IOException
+   */
+  @Override
+  public void addTransaction(long containerID, List<Long> blocks)
+      throws IOException {
+    lock.lock();
+    try {
+      // TODO(runzhiwang): Should use distributed sequence id generator
+      Long nextTXID = UniqueId.next();
+      DeletedBlocksTransaction tx =
+          constructNewTransaction(nextTXID, containerID, blocks);
+
+      StorageContainerDatanodeProtocolProtos.DeleteBlocksCommandProto proto =
+          StorageContainerDatanodeProtocolProtos.DeleteBlocksCommandProto
+              .newBuilder()
+              .addDeletedBlocksTransactions(tx).setCmdId(-1)
+              .build();
+
+      deletedBlockLogStateManager.addTransactionsToDB(proto);
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  @Override
+  public int getNumOfValidTransactions() throws IOException {
+    lock.lock();
+    try {
+      final AtomicInteger num = new AtomicInteger(0);
+      try (TableIterator<Long,
+          ? extends Table.KeyValue<Long, DeletedBlocksTransaction>> iter =
+               deletedBlockLogStateManager.getReadOnlyIterator()) {
+        while (iter.hasNext()) {
+          DeletedBlocksTransaction delTX = iter.next().getValue();
+          if (delTX.getCount() > -1) {
+            num.incrementAndGet();
+          }
+        }
+      }
+      return num.get();
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @param containerBlocksMap a map of containerBlocks.
+   * @throws IOException
+   */
+  @Override
+  public void addTransactions(Map<Long, List<Long>> containerBlocksMap)
+      throws IOException {
+    lock.lock();
+    try {
+      List<DeletedBlocksTransaction> txs = new ArrayList<>();
+      for (Map.Entry< Long, List< Long > > entry :
+          containerBlocksMap.entrySet()) {
+        // TODO(runzhiwang): Should use distributed sequence id generator
```

Comment:
+1

I am waiting https://github.com/apache/ozone/pull/1737 before sending the ID gen PR.